todo_app/test_app.py

```diff
@@ -0,0 +1,85 @@
+import unittest
+from todo_app.trello_items import Item
+from todo_app.app import ViewModel
+    
+
+
+def test_to_do_items_only_shows_items_in_the_to_do_list():
+    # Test Case Setup
+    items = []
+
+    to_do_item = Item("1", "ToDoItemName", "To Do")
+    items.append(to_do_item)
+
+    doing_item = Item("2", "DoingItemName", "Doing")
+    items.append(doing_item)
+
+    done_item = Item("3", "DoneItemName", "Done")
+    items.append(done_item)
+    
+    view_model = ViewModel(items)
+
+    # End Test Case Setup
+
+    # Start Testing...
+    result: list[Item] = view_model.to_do_items
+
+    # Checking everything worked
+    assert len(result) == 1
+    item = result[0]
+    assert item.status == "To Do"
+    assert result == [to_do_item] 
+
+
+def test_doing_items_only_shows_items_in_the_doing_list():
+    # Test Case Setup
+    items = []
+
+    to_do_item = Item("1", "ToDoItemName", "To Do")
+    items.append(to_do_item)
+
+    doing_item = Item("2", "DoingItemName", "Doing")
+    items.append(doing_item)
+
+    done_item = Item("3", "DoneItemName", "Done")
+    items.append(done_item)
+    
+    view_model = ViewModel(items)
+
+    # End Test Case Setup
+
+    # Start Testing...
+    result: list[Item] = view_model.doing_items
+
+    # Checking everything worked
+    assert len(result) == 1
+    item = result[0]
+    assert item.status == "Doing"
+    assert result == [doing_item] 
+
+def test_done_items_only_shows_items_in_the_done_list():
+    # Test Case Setup
```

Comment:
Nice clear tests here!
Your setup has a fair amount of duplication; we could remove that either by:
* Declaring the items list outside the functions, and sharing it between them
* Or even nicer, converting it to a pytest fixture that they can share; that way they can't accidentally interfere with each other

For example, we might extract it like:
```python

@pytest.fixture
def items():
    _items = []
    to_do_item = Item("1", "ToDoItemName", "To Do")
    _items.append(to_do_item)
    doing_item = Item("2", "DoingItemName", "Doing")
    _items.append(doing_item)
    done_item = Item("3", "DoneItemName", "Done")
    _items.append(done_item)
    return _items

def test_to_do_items_only_shows_items_in_the_to_do_list(items):
    # Test Case Setup    
    view_model = ViewModel(items)

    # End Test Case Setup

    # Start Testing...
    result: list[Item] = view_model.to_do_items

    # Checking everything worked
    assert len(result) == 1
    item = result[0]
    assert item.status == "To Do"
    assert item.name == 'ToDoItemName'

```

Which would then allow most of the setup to be shared with the other two tests

Reply:
Done